### PR TITLE
Task done

### PR DESCRIPTION
**Pull request by AntonBaranskyi**
https://antonbaranskyi.github.io/react_todo-app-with-api/

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -8,5 +8,6 @@ module.exports = {
     'jsx-a11y/label-has-associated-control': ["error", {
       assert: "either",
     }],
+    'import/no-cycle': 'off',
```
Comment (vlopatko) [inline .eslintrc.js:11]: I think these changes are not good, cause you import files by cycle, and do not need to disable these eslint rules.

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    [todos],
+  );
+
+  const countFinished = useCallback(
+    (listTodo: Todo[]) => {
+      return listTodo.map((todo) => todo.completed).filter((todo) => todo)
+        .length;
+    },
+    [todos],
+  );
```
Comment (vlopatko) [inline src/components/Footer/Footer.tsx:23]: You can move these functions to a utils file or something like utils, or just out of the component, and useCallback will be do not need

File: src/components/TodoItem/TodoItem.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        <div className="modal overlay">
+          <div className="modal-background has-background-white-ter" />
+          <div className="loader" />
+        </div>
+      )}
+
+      <div className="modal overlay">
+        <div className="modal-background has-background-white-ter" />
+        <div className="loader" />
+      </div>
```
Comment (vlopatko) [inline src/components/TodoItem/TodoItem.tsx:139]: You can use ClassNames library for editing classname , and you will not be needed to create 3 same elements.
Follow DRY rules

**Review by vlopatko — APPROVED** [review]
Good job! 👍🏻 Looks nice 💯 
But I have some advices for you, please check my comments.

File: src/components/AppBody/AppBody.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { Footer } from '../Footer';
+import { ErrorComponent } from '../ErrorComponent';
+import { Header } from '../Header';
+import { filterTodo } from '../../helpers/filterTodo';
+import { TempTodoItem } from '../TempTodoItem';
+
+export const AppBody: React.FC = () => {
+  const { todos, sortField, tempTodo } = useContext(TodosContext);
+
+  const visibleTodos = filterTodo(todos, sortField);
```
Comment (anstsot) [inline src/components/AppBody/AppBody.tsx:13]: use `useMemo` hook

File: src/components/Footer/Footer.tsx (hunk 2 above)
Comment (anstsot) [inline src/components/Footer/Footer.tsx:23]: better to use `useMemo`

File: src/components/Header/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+export const Header: React.FC = () => {
+  const { toggleAll } = useContext(TodosContext);
+
+  return (
+    <header className="todoapp__header">
+      <button
+        type="button"
+        className="todoapp__toggle-all active"
+        onClick={() => toggleAll()}
```
Comment (anstsot) [inline src/components/Header/Header.tsx:14]:
```suggestion
        onClick={toggleAll}
```

**Review by anstsot — CHANGES_REQUESTED** [review]
There are few `eslint` issues (fix them) and let's fix few bugs:
1. It's possible to change the todo to empty 
image

2. By clicking on toggle the loaders have all todos (even completed)
image

3. there is loader and error on remove (the loades shouldn't be visible on error)
image

4. There is an issue with focus on editing

**Review by anstsot — CHANGES_REQUESTED** [review]
2 first bugs are still reproducible. check them, please
Fix `eslint` issue
image

**Pull request closed without merging**